cranelift/filetests/src/test_run.rs

```diff
@@ -42,8 +42,22 @@ impl SubTest for TestRun {
                 let command = parse_run_command(trimmed_comment, &func.signature)
                     .map_err(|e| format!("{}", e))?;
                 trace!("Parsed run command: {}", command);
-                // TODO in following changes we will use the parsed command to alter FunctionRunner's behavior.
 
+                // If this test requests to run on a completely different
+                // architecture than the host platform then we skip it entirely,
+                // since we won't be able to natively execute machine code.
+                let requested_arch = context.isa.unwrap().triple().architecture;
+                if requested_arch != Architecture::host() {
+                    return Ok(());
+                }
+
+                // TODO in following changes we will use the parsed command to alter FunctionRunner's behavior.
+                //
+                // Note that here we're also explicitly ignoring `context.isa`,
+                // regardless of what's requested. We want to use the native
+                // host ISA no matter what here, so the ISA listed in the file
+                // is only used as a filter to not run into situations like
+                // running x86_64 code on aarch64 platforms.
```

Comment:
See #1517 for where I remove this TODO and finish this implementation. I agree that we need a way to filter out `test run` filetests based on the target ISA but I don't want to confuse anyone by overloading `target ...`; perhaps if we add some documentation at the top of this file (for cargo docs) and some text to the bottom of https://github.com/bytecodealliance/wasmtime/blob/master/cranelift/docs/testing.md?

Reply:
To make this even more explicit we could add another keyword `only_run_on` (or something like that), but that could be a separate issue/PR.

Reply:
I'm fine basically doing whatever here, although I'm pretty unfamiliar with cranelift's currently test runner/suite. It seems like documenting this change is a bare minimum, but beyond that do you think something should happen here or in a future pr?

Reply:
Hm, I don't feel really strongly about the `only_run_on` idea so I guess we could just document and if someone else cares enough we could take it further.